src/datepicker/datepicker.js

```diff
@@ -535,6 +536,30 @@ function($compile, $parse, $document, $rootScope, $position, dateFilter, datePar
         return (new Date(date1.getFullYear(), date1.getMonth(), date1.getDate()) - new Date(date2.getFullYear(), date2.getMonth(), date2.getDate()));
       };
 
+      if(timelessJsonMode) {
+        scope.jsonToDate = function(date) {
+          if (date && angular.isString(date) && date.indexOf('-') > -1) {
+            var update = date.indexOf('T') > -1;
+            date = date.substring(0, 10).split('-');
+            date = new Date(+date[0], +date[1] - 1, +date[2]);
+            if (update) {
+              ngModel.$setViewValue(dateFilter(date, dateFormat));
+              ngModel.$render(); 
+            }
+          }
+          return date;
+        };
+        
+        scope.$parent.jsonToTicks = function(date) {
```

Comment:
@wesleycho Yeah, that was nagging me too. I plan to fix this. When I go `scope.prop = function` it isn't available on the element scope. Do you happen to have any advice here: any special way of adding to this scope in a way that persists past the link function?